app.rb

```diff
@@ -0,0 +1,162 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'rental'
+
+# Initialize arrays to keep track of all book and person instances
+books = []
+people = []
+
+# Define the main entry point for the console app
+def main(books, people)
+  puts 'Welcome to the Library Console App!'
+
+  loop do
+    puts "\nPlease choose an option:"
+    puts '1. List all books'
+    puts '2. List all people'
+    puts '3. Create a person (teacher or student)'
+    puts '4. Create a book'
+    puts '5. Create a rental'
+    puts '6. List all rentals for a given person id'
+    puts '7. Quit the app'
+
+    choice = gets.chomp.to_i
+
+    case choice
+    when 1
+      list_all_books(books)
+    when 2
+      list_all_people(people)
+    when 3
+      create_person(people)
+    when 4
+      create_book(books)
+    when 5
+      create_rental(books, people)
+    when 6
+      list_rentals_for_person(people)
+    when 7
+      puts 'Exiting the app. Goodbye!'
+      break
+    else
+      puts 'Invalid choice. Please try again.'
+    end
+  end
+end
+
+def list_all_books(books)
+  puts "\nList of all books:"
+  books.each do |book|
+    puts "#{book.title} by #{book.author}"
+  end
+end
+
+def list_all_people(people)
+  puts "\nList of all people:"
+  people.each do |person|
+    puts "#{person.name} (#{person.class.name}) - ID: #{person.id}"
+  end
+end
+
+def create_person(people)
+  puts "\nCreate a person:"
+  puts "Enter the person's name:"
+  name = gets.chomp
+
+  puts "Enter the person's age:"
+  age = gets.chomp.to_i
+
+  puts 'Is it a teacher or a student? (T/S)'
+  role = gets.chomp.downcase
```

Comment:
- [ ] As shown in the demo, this feature is supposed to come before entering the name of the person, which means before typing the name of the person one should know if they are a student or teacher, Kindly make sure you correct for a better User Experience.